_posts/2017-07-07-tech-behind-dta-website.md

```diff
@@ -0,0 +1,100 @@
+
+Our website is a Commonwealth record and needs to be managed in accordance with the Archives Act 1983 and the [National Archives of Australia guidance](http://www.naa.gov.au/Images/archweb_guide_tcm16-47165.pdf). Since every change to the site is tracked in version control, we can easily meet this requirement by going back and rebuilding the site as it was at any point in time.
+
+## Hosting our site — cloud.gov.au
+
+The DTA website is hosted on [cloud.gov.au](https://www.dta.gov.au/what-we-do/platforms/cloud/) – the platform the DTA is developing to make it easier for government to build digital services. Cloud.gov.au saves time and effort by managing infrastructure, so a team can focus on building the service.
+
+Cloud.gov.au has an API which allows deployments to be entirely automated. We specify the [static build pack](http://docs.cloudfoundry.org/buildpacks/staticfile/index.html) in our [manifest file](https://github.com/AusDTO/dta-website/blob/develop/manifest-production.yml), and the platform takes care of provisioning a server and inserting the built website files. This means our exposure to security attacks is minimised, and we do not have to spend time regularly applying platform updates.
+
+We use [Cloudfront](https://github.com/AusDTO/dta-website/blob/develop/manifest-production.yml) as our content delivery network, and requests are routed to a HTTPS termination layer on cloud.gov.au before being forwarded to our web server.
+
+## Testing
+
+The build pipeline is a series of scripts, which allows us to customise the process and add tools to check the quality of the site. We run [HTMLProofer](https://github.com/gjtorikian/html-proofer) which quickly validates the static files HTML, and makes sure there are no broken internal links or missing images.
+
+We also have ad-hoc tests that run on a local copy of the site. We can:
+- check for broken external links
+- test all pages for obvious accessibility using [pa11y-ci](https://github.com/pa11y/ci) (recognising we still need to undertake manual testing)
+
+We have found these are best kept out of CI — we want our CI to be fast, and having build dependencies on every external server puts us at risk of outages. 
+
+## External services and plugins
+
+A static site on its own has limitations, but this is where external services are useful. The ones we currently use are:
+- [Disqus](https://disqus.com/) for our blog commenting system
+- [Youtube](https://www.youtube.com/channel/UCmDkFN3UlK2wSKDQQhd-Y-A) for hosting our videos
+- [Recruiterbox](https://recruiterbox.com/) for showing [our open jobs](https://www.dta.gov.au/who-we-are/corporate/jobs/)
+- [Google Apps Script](https://www.google.com/script/start/) for receiving form submissions
+
+Site search is something you might expect to be an external service, but we use [client-side JavaScript](https://www.google.com/script/start/). We could improve our search with an external service, but for our current usage this is free and fast.
```

Comment:
This link should be to https://github.com/slashdotdash/jekyll-lunr-js-search - this is the jekyll plugin we use which creates the search javascript and index at build-time (Well actually we use a slightly modified local version, but let's not complicate things!)